[Level 1 - Part 2] - Setup recipe pages

jeffgaynor opened the pull request:
- Added recipe page content
- Added recipe page styles
- Swapped @next/mdx for next-mdx-remote
- Refactored recipe page loading with dynamic routing
- Added documentation

**Level 1 ready:**

<img width="1330" alt="Screen Shot 2022-04-01 at 9 38 09 AM" src="https://user-images.githubusercontent.com/508871/161285642-f667ca49-bf96-494f-9e2b-fca7e731eba5.png">

jeffgaynor committed:
[Level 1 - Part 2] - Setup recipe pages

- Added recipe page content
- Added recipe page styles
- Swapped @next/mdx for next-mdx-remote
- Refactored recipe page loading with dynamic routing
- Added documentation

kimlisa commented on pages/recipes/[slug].tsx:
```diff
@@ -0,0 +1,129 @@
+  );
+};
+
+export const getStaticProps = async ({ params: { slug } }: Params) => {
+  const markdown = fs.readFileSync(`_recipes/${slug}.md`);
+  const source = await serialize(markdown.toString(), { parseFrontmatter: true });
+
+  return {
+    props: {
+      source,
+    },
+  };
+};
+
+export const getStaticPaths = () => ({
```
there should be a type `GetStaticProps` and `GetStaticPaths` provided by next

jeffgaynor replied:
I'll take a look. Thanks.

C-STYR reviewed (approved):
Looks good, Jeff!
Some general comments from Alexey and me (no need to respond directly - just points to consider for future PRs):
- there still seems to be some non-relevant boilerplate 
- `svgo.config.js` is throwing errors for props that are not part of `preset`
- `Theme.ts` contains only color themes, not spacing or font-size
- If the app were to scale to a hundreds of recipes (and CMS was NOT being integrated), how would assets in `public/images` need to be managed?
- `Head` component is static - all recipes will have duplicate title, meta content
- `Head` component contains no social meta content
- SSR not yet fully enabled with Styled-Components - key piece missing
- styling tags as children (`ul`, `li` in `nav` for example) can lead to problems down the road - what is a more scalable approach?

jeffgaynor commented:
Thanks for the feedback!

> Looks good, Jeff! Some general comments from Alexey and me (no need to respond directly - just points to consider for future PRs):
> 
> * there still seems to be some non-relevant boilerplate
Can you point to a spot or two in the code, so I know to what you are referring?

> * `svgo.config.js` is throwing errors for props that are not part of `preset`
👍 I already removed that in the PR I have in waiting.

> * `Theme.ts` contains only color themes, not spacing or font-size
👍 I already removed that in the PR I have in waiting.

> * If the app were to scale to a hundreds of recipes (and CMS was NOT being integrated), how would assets in `public/images` need to be managed?
In general, it would be much better to deploy the images separately to a CDN. As I'm new to Next, I'm not quite sure how that plays with it's image optimization. What would you recommend?

> * `Head` component is static - all recipes will have duplicate title, meta content
👍 I have this updated in the PR I have in waiting.

> * `Head` component contains no social meta content
That would be nice to have for sure. Due to time constraints, I didn't go to the level of building out every aspect - I stayed focused on meeting the key requirements of the challenge.

> * SSR not yet fully enabled with Styled-Components - key piece missing
Are you talking about using a ServerStyleSheet? https://styled-components.com/docs/advanced#example

> * styling tags as children (`ul`, `li` in `nav` for example) can lead to problems down the road - what is a more scalable approach?
It depends on the use case. Are you referring to the navigational elements, or the ingredients list? What problems specifically?

C-STYR commented:
> Thanks for the feedback!
> 
> > Looks good, Jeff! Some general comments from Alexey and me (no need to respond directly - just points to consider for future PRs):
> > 
> > * there still seems to be some non-relevant boilerplate
> >   Can you point to a spot or two in the code, so I know to what you are referring?

--`pages/api`, potentially `globals.css`, anything else that came in with the `create-next-app` package that isn't being used
> 
> > * `svgo.config.js` is throwing errors for props that are not part of `preset`
> >   👍 I already removed that in the PR I have in waiting.
> 
> > * `Theme.ts` contains only color themes, not spacing or font-size
> >   👍 I already removed that in the PR I have in waiting.
> 
> > * If the app were to scale to a hundreds of recipes (and CMS was NOT being integrated), how would assets in `public/images` need to be managed?
> >   In general, it would be much better to deploy the images separately to a CDN. As I'm new to Next, I'm not quite sure how that plays with it's image optimization. What would you recommend?

-- Vercel has a CDN built in, this was more of an organizational comment. A clear link between images and individual recipes within the file structure might make it easier to manage recipes in general (for instance if one were removing recipes as often as adding them).
> 
> > * `Head` component is static - all recipes will have duplicate title, meta content
> >   👍 I have this updated in the PR I have in waiting.
> 
> > * `Head` component contains no social meta content
> >   That would be nice to have for sure. Due to time constraints, I didn't go to the level of building out every aspect - I stayed focused on meeting the key requirements of the challenge.
> 
> > * SSR not yet fully enabled with Styled-Components - key piece missing
> >   Are you talking about using a ServerStyleSheet? https://styled-components.com/docs/advanced#example

--yes!
> 
> > * styling tags as children (`ul`, `li` in `nav` for example) can lead to problems down the road - what is a more scalable approach?
> >   It depends on the use case. Are you referring to the navigational elements, or the ingredients list? What problems specifically?

--further development of the `Root` styled.nav component in either `Nav.tsx` or `SideNav.tsx`, for example, additional nesting levels, might lead to scoping issues.  Styling class names vs. tags would prevent this.

jeffgaynor commented:
@C-STYR Thanks for the additional answers, that's helpful! Given more time, I would definitely have done several of the things you mentioned. We're on the same page!
